packages/ui/src/lib/button/Button.svelte

```diff
@@ -1,3 +1,49 @@
-<button class="btn bg-core-blue-600 p-4">
-    <slot></slot>
-</button>
+<script lang="ts">
+	import classNames from 'classnames';
+
+	export let style: 'primary' | 'secondary' | 'ghost' = 'primary';
+	export let size: 'xs' | 'sm' | 'base' | 'lg' | 'xl' = 'base';
+	export let href: string | undefined = undefined;
+	export let type: 'button' | 'submit' = 'button';
+
+	const styleClasses = {
+		primary: 'bg-core-grey-500 border-b-4 border-core-blue-600 text-white hover:bg-core-grey-400',
+		secondary: 'bg-white border-b-4 border-core-blue-600 hover:bg-core-grey-100',
+		ghost: 'bg-transparent hover:bg-core-grey-100 dark:text-white dark:hover:bg-core-grey-400'
+	};
+
+	const sizeClasses = {
+		xs: 'text-xs',
+		sm: 'text-sm',
+		base: 'text-base',
+		lg: 'text-lg',
+		xl: 'text-xl'
+	};
+
+	let buttonClass: string;
+
+	$: buttonClass = classNames(
+		'px-4 py-2 inline-flex justify-center',
+		sizeClasses[size],
+		styleClasses[style],
+		$$props.disabled && 'cursor-not-allowed text-core-grey-300 border-core-grey-200 opacity-50',
+		$$props.class
+	);
+</script>
+
+<svelte:element
+	this={href ? 'a' : 'button'}
+	type={href ? undefined : type}
+	class={buttonClass}
+	on:click
+	on:change
+	on:keydown
+	on:keyup
+	on:touchstart
+	on:touchend
+	on:touchcancel
+	on:mouseenter
+	on:mouseleave
```

Comment:
**Just a note and a minor thing:** Event forwarding is nice but I've found it hides accessibility and A11y warnings. I'm not sure what to do about as yet.

Then again, the Svelte devs may be sorting it out (or perhaps they already have and I haven't noticed).